### ISSUE TITLE
Undefined reference to `OSSL_set_max_threads`

### ISSUE THREAD

**Issue by Lutymane** [openssl/openssl#21305]
Hello,

I built openssl from master branch (with `./Configure no-deprecated no-apps`) and installed it (`sudo make install_sw`). But I'm having a problem with compiling a program which has openssl as dependency. If I use `OSSL_set_max_threads`, it will fail with 

```
/usr/bin/ld: /tmp/cc0eDR8y.o: in function `main':
main.cpp:(.text+0x16c): undefined reference to `OSSL_set_max_threads(ossl_lib_ctx_st*, unsigned long)'
collect2: error: ld returned 1 exit status
```

If i remove usage of that function from the code, it does compile and everything else from openssl works properly. In `configuration.h` macro `OPENSSL_THREADS` is being defined. So it does seem to have multithreading enabled. But I can't understand what's the problem.

**Comment by paulidale** [comment]
I suspect you are picking up the system OpenSSL not the one you installed.

**Comment by paulidale** [comment]
It doesn't look like you are getting far enough to know if Argon2 is available or not.
Direct calls to Argon2 don't exist in OpenSSL's public API -- EVP_KDF_fetch() doesn't know about Argon2 directly e.g.

**Comment by Lutymane** [comment]
@paulidale what I meant is my program works with Argon2 using 1 thread, but I can't use more threads because of the issue with `OSSL_set_max_threads`

**Comment by t8m** [comment]
Now I see what's the issue - the header file is missing the extern "C" clause.

**Comment by t8m** [comment]
Fix in #21307
